keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -402,182 +393,262 @@ The following new `AdminNetworkPolicy` API will be added to the `policy.networki
+  // Ingress and Egress rules in a single AdminNetworkPolicy Instance. 
   Egress      []AdminNetworkPolicyEgressRule
 }
 
-// Exactly one of the selector pointers should be set in a given AppliedTo
-type AppliedTo struct {
+// AdminNetworkPolicySubject defines what objects the policy selects. 
+// Exactly one of the selector pointers should be set.  
+type AdminNetworkPolicySubject struct {
   NamespaceSelector         *metav1.LabelSelector
   NamespaceAndPodSelector   *NamespaceAndPodSelector
 }
 
-// NamespaceAndPodSelector allows the user to select a namespace/s and optionally 
-// select a given set of pods in that namespace/s
+// NamespaceAndPodSelector allows the user to select a Namespace(s) and optionally 
+// select a given set of pod(s) in that namespace(s)
 type NamespaceAndPodSelector struct {
+  // This field follows standard label selector semantics; if present but empty, 
+  // it selects all Namespaces.  
   NamespaceSelector   metav1.LabelSelector
-  // Used to explicitly select pods within a namespace
-  PodSelector         *metav1.LabelSelector
+
+  // Used to explicitly select pods within a namespace; if present but empty, 
+  // it selects all Pods.  
+  PodSelector         metav1.LabelSelector
 }
 
+// AdminNetworkPolicyIngressRule describes an action to take on a particular 
+// set of traffic destined for pods selected by an AdminNetworkPolicy's 
+// Subject field. The traffic must match both l4Selector and from. 
 type AdminNetworkPolicyIngressRule struct {
-  // Name is an identifier for this rule. Must be unique within an instance of ANP. Can be
-  // same across policy instances.
+  // Name is an identifier for this rule.
   // +optional
-  Name      string
+  Name         string
+
   // Action specifies whether this rule must pass, allow or deny traffic. 
-  // Allow: explicitly allows the selected traffic 
-  // Deny: explicitly denies the selected traffic
-  // Pass: explicitly allows the selected traffic to skip and remaining ANP rules 
+  // Allow: allows the selected traffic 
+  // Deny: denies the selected traffic
+  // Pass: allows the selected traffic to skip and remaining positive priority ANP rules 
   // and be delegated by K8's Network Policy. 
-  Action    RuleAction
-  // List of ports for incoming traffic.
-  // Each item in this list is combined using a logical OR. If this field is
-  // empty or missing, this rule matches no ports (traffic not restricted by port).
-  // If this field is present and contains at least one item, then this rule 
-  // allows/denies/passes traffic only if the traffic matches at least one port in the list.
-  // +optional
-  Ports     []networkingv1.NetworkPolicyPort
-	// List of sources from which traffic will be allowed/denied/passed to the entities 
+  Action       AdminNetPolRuleAction
+  
+  // L4Selector allows for matching traffic based on L4 constructs. 
+  L4Selector   AdminNetworkPolicyL4Selector
```

Comment:
We wanted to simply call it ports, butt the problem is that if user does not elect `allPorts` and indeed want to do port selection, then the yaml ends up like
```
ports:
  ports:
  - protocol: TCP
     port: 80
```
even though with allports it sounds just right
```
ports:
  allPorts: true
```
CNI implementations often call ports+protocol "service", would be make sense here?, i.e.
```
ports:
  services:
  - protocol: TCP
     port: 80
```
happy to hear a better name though...

Reply:
```yaml
ports:
  - protocol: TCP
    port: 443
  - protocol: SCTP
    allPorts: true
```

what about that?

Reply:
@squeed I thought about this as well, but then the user has to specify the protocol(even if they have no idea what it means) in that case I feel like they really would just want to specify all-ports all-protocols. And if they _did_ want to specify the protocol they could utilize port-range to express "all ports"

Reply:
Hmm. What about allowing wildcards then? A protocol of `*` or `ANY` matches all protocols (and thus specifying a port isn't allowed). Likewise, a port of `*` matches all ports for a given protocol.

Reply:
I don't hate this, but maybe because I know the rationale:
```
ports:                  ports:
  ports:                  allPorts: true
    - protocol: TCP
      port: 443
```

What about:
```
ports:                  ports:
  list:                   allPorts: true
    - protocol: TCP
      port: 443
```

meh.

I don't HATE a wildcard, but it leaves us with a case of "either (allPorts) OR (port AND MAYBE endPort)", which usually ends in tears, except it seems unlikely to grow a need way to spec ports (famous last words).
```
ports:                  ports:
  - protocol: TCP         - protocol: *
    port: 443               allPorts: true
```

That's not far from:
```
ports:                  ports:
  - protocol: TCP         - protocol: *
    port: 443               ports: 1
                            endPort: 65535
```

For the same of KEP freeze, pick one and we can revisit in API review.